### ISSUE TITLE
Breaking change in most recent version of package:test ? 

### ISSUE THREAD

**Issue by mkustermann** [dart-lang/test#515]
Could the most recent version of package test have included a breaking change? A lot of our builders running package tests on the [package waterfall](https://build.chromium.org/p/client.dart.packages/console) started suddenly to fail. For example from [running package:async tests](https://build.chromium.org/p/client.dart.packages/builders/packages-linux-async/builds/859/steps/pub%20run%20test/logs/stdio):

```
...
00:01 +0 -1: [Chrome] test/typed_wrapper/stream_subscription_test.dart: with valid types, forwards onData()
  Invalid argument(s): The wrapped function has more than 6 required arguments
  package:test                                            Object.expectAsync
  test/typed_wrapper/stream_subscription_test.dart 26:22  main.<fn>.<fn>
  package:stack_trace                                     StackZoneSpecification._registerBinaryCallback.<fn>
...
```

@nex3 Maybe you have an idea?

**Comment by kevmoo** [comment]
Just hit this w/ pkg/shelf - https://travis-ci.org/dart-lang/shelf/jobs/185835897#L701

I patched in Natalie's fix – https://github.com/dart-lang/test/pull/514 – and we're looking better

https://travis-ci.org/dart-lang/shelf/builds/185839810

But now I'm hitting https://github.com/dart-lang/sdk/issues/28045

**Comment by maxwellpeterson-wf** [comment]
Hitting this same issue on a lot of our libraries at Workiva.

**Comment by kevmoo** [comment]
@maxwellpeterson-wf – we'll get @nex3 patch in ASAP.

Are you seeing dart-lang/sdk#28045, too?

**Comment by maxwellpeterson-wf** [comment]
@kevmoo no, I haven't seen that yet. I don't think we have addressed the `expectAsync` deprecation in many places yet, so perhaps that is why.

**Comment by kevmoo** [comment]
This is merged – https://github.com/dart-lang/test/pull/514 – as is this – https://github.com/dart-lang/sdk/commit/7c3c297a8ad907f12bf12b96a21777421067fd18 into 1.22.0-dev.4

All should be good now – I think/hope.

**Comment by kevmoo** [comment]
> This is merged – #514 – as is this – dart-lang/sdk@7c3c297 into 1.22.0-dev.4

Confirmed. Closing.
